src/fileAttachment.js

```diff
@@ -12,9 +13,16 @@ async function remote_fetch(file) {
 
 async function dsv(file, delimiter, {array = false, typed = false} = {}) {
   const text = await file.text();
-  return (delimiter === "\t"
-      ? (array ? tsvParseRows : tsvParse)
-      : (array ? csvParseRows : csvParse))(text, typed && autoType);
+  const parse = (delimiter === "\t"
+    ? (array ? tsvParseRows : tsvParse)
+    : (array ? csvParseRows : csvParse));
+  if (typed === "auto") {
```

Comment:
```suggestion
  if (typed === "auto" && !array) {
```

coerceRow isn’t compatible with `array`; it’ll generate an object for each row, but we’re supposed to return an array-of-arrays. I guess we could change coerceRow to support generating arrays instead of objects, but I’m not sure it really makes sense to run inferSchema on the array case, because the array case is generally for when the CSV file has an unusual shape (_e.g._, a variable number of fields per line).